src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -236,4 +227,26 @@ private static boolean checkIsAllLinesAreWhitespace(String[] lines, int lineFrom
         }
         return result;
     }
+
+    /**
+     * Calculates the left curly corresponding to the block to be checked.
+     *
+     * @param ast a {@code DetailAST} value
+     * @return the left curly corresponding to the block to be checked
+     */
+    private static DetailAST findLeftCurly(DetailAST ast) {
+        DetailAST leftCurly;
+        if (ast.findFirstToken(TokenTypes.SLIST) == null) {
+            leftCurly = ast.findFirstToken(TokenTypes.LCURLY);
+        }
+        else {
+            leftCurly = ast.findFirstToken(TokenTypes.SLIST);
+        }
+        if (ast.getType() == TokenTypes.LITERAL_CASE
+                && ast.getNextSibling() != null
+                && "{".equals(ast.getNextSibling().getFirstChild().getText())) {
```

Comment:
@ps-sp 
@romani's approach will work. I think that you need to find a right curly in visit token in separate method. The same as you do in findRightCurly. The argument of hasText is incorrect to y mind. We analyze code block, not just SLIST, so we need to find the block boundaries. hasText should receive two parameters: left curly which opens a block and right curly which closes the bock. When you find both right curly and left curly you will be able to check whether they are null and pass them into hasText.